src/frontend/lib.nim

```diff
@@ -421,6 +421,34 @@ when defined(ctIndex) or defined(ctTest):
     # for more info, please read the comment for `setupLdLibraryPath` there
     nodeProcess.env["LD_LIBRARY_PATH"] = nodeProcess.env["CT_LD_LIBRARY_PATH"]
 
+  proc runUploadWithStreaming*(
+      path: cstring,
+      args: seq[cstring],
+      onData: proc(data: string),
```

Comment:
we could use `cstring` here, but it can work like that as well, maybe with more than needed string conversions